### ISSUE TITLE
unit test event handlers

### ISSUE THREAD

**Issue by boudra** [gitcoinco/grants-stack-indexer#114]
The switch is big and it's hard to follow. We can put event handlers into functions and calling those functions from the switch.

See `matchAmountUpdated` as an example.

**Comment by bard** [comment]
An important goal of this is to improve testability by having smaller, well-defined units of functionality.

**Comment by bard** [comment]
poc: https://github.com/gitcoinco/allo-indexer/pull/161
